src/ralph/backends/data/es.py

```diff
@@ -197,20 +192,20 @@ def list(
         for index in indices:
             yield index
 
-    @enforce_query_checks
     def read(  # noqa: PLR0912, PLR0913
         self,
         *,
-        query: Optional[Union[str, ESQuery]] = None,
+        query: Optional[ESQuery] = None,
         target: Optional[str] = None,
         chunk_size: Optional[int] = None,
         raw_output: bool = False,
         ignore_errors: bool = False,
     ) -> Iterator[Union[bytes, dict]]:
+        # pylint: disable=too-many-arguments,too-many-branches
         """Read documents matching the query in the target index and yield them.
 
         Args:
-            query (str or ESQuery): A query in the Lucene query string syntax or a
+            query (ESQuery): A query in the Lucene query string syntax or a
```

Comment:
So from now on, we can't pass a string in the Lucene query syntax, right?
Fine with it but we should change the docstrings